File: server/loans/api/src/main/java/ee/rsx/kata/bank/loans/api/eligibility/LoanEligibilityResultDTO.java
package ee.rsx.kata.bank.loans.api.eligibility;

import java.util.List;

public record LoanEligibilityResultDTO(
  LoanEligibilityStatus result,
  List<String> errors,
  String ssn,
  Integer loanAmount,
  Integer loanPeriodMonths,
  Integer eligibleLoanAmount,
  Integer eligibleLoanPeriod
) {}

File: server/loans/api/src/main/kotlin/ee/rsx/kata/bank/loans/eligibility/LoanEligibilityResultDTO.kt
package ee.rsx.kata.bank.loans.eligibility

data class LoanEligibilityResultDTO(
  val result: LoanEligibilityStatus,
  val errors: List<String>? = null,
  val ssn: String,
  val loanAmount: Int,
  val loanPeriodMonths: Int,
  val eligibleLoanAmount: Int? = null,
  val eligibleLoanPeriod: Int? = null
)
